src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -6804,283 +6913,207 @@ namespace WaterThermalTanks {
+					if ( HeatPump.HPSetPointError == 1 ) {
+						ShowWarningError( HeatPump.Type + " \"" + HeatPump.Name + ":  Water heater tank set point temperature is greater than or equal to the cut-in temperature of the heat pump water heater. Heat Pump will be disabled and simulation continues." );
 						ShowContinueErrorTimeStamp( " ...Heat Pump cut-in temperature=" + HPMinTempChar );
 					} else {
-						ShowRecurringWarningErrorAtEnd( HPWaterHeater( HPNum ).Type + " \"" + HPWaterHeater( HPNum ).Name + ":  Water heater tank set point temperature is greater than or equal to the cut-in temperature of the heat pump water heater. Heat Pump will be disabled error continues...", HPWaterHeater( HPNum ).HPSetPointErrIndex1, HPMinTemp, HPMinTemp );
+						ShowRecurringWarningErrorAtEnd( HeatPump.Type + " \"" + HeatPump.Name + ":  Water heater tank set point temperature is greater than or equal to the cut-in temperature of the heat pump water heater. Heat Pump will be disabled error continues...", HeatPump.HPSetPointErrIndex1, HPMinTemp, HPMinTemp );
 					}
 				}
 			}
 			return;
 		}
-		{ auto const SELECT_CASE_var( HPWaterHeater( HPNum ).TankTypeNum );
+		{ auto const SELECT_CASE_var( HeatPump.TankTypeNum );
 		if ( SELECT_CASE_var == MixedWaterHeater ) {
-			TankTemp = WaterThermalTank( WaterThermalTankNum ).SavedTankTemp;
+			TankTemp = Tank.SavedTankTemp;
 		} else if ( SELECT_CASE_var == StratifiedWaterHeater ) {
-			TankTemp = FindStratifiedTankSensedTemp( WaterThermalTankNum, HPWaterHeater( HPNum ).ControlSensorLocation );
+			TankTemp = FindStratifiedTankSensedTemp( WaterThermalTankNum, HeatPump.ControlSensorLocation );
 		} else {
 			assert( false );
 		}}
-		HPWaterHeater( HPNum ).Mode = HPWaterHeater( HPNum ).SaveMode;
+		HeatPump.Mode = HeatPump.SaveMode;
 
 		// set the heat pump air- and water-side mass flow rate
-		MdotWater = HPWaterHeater( HPNum ).OperatingWaterFlowRate * RhoH2O( TankTemp );
+		MdotWater = HeatPump.OperatingWaterFlowRate * RhoH2O( TankTemp );
 
-		//     select mode of operation (float mode or heat mode)
-		{ auto const SELECT_CASE_var( HPWaterHeater( HPNum ).Mode );
-		//       HPWH was heating last iteration and will continue to heat until the set point is reached
+		// Select mode of operation (float mode or heat mode) from last iteration.
+		// Determine if heating will occur this iteration and get an estimate of the PLR
+		{ auto const SELECT_CASE_var( HeatPump.Mode );
 		if ( SELECT_CASE_var == HeatMode ) {
-
+			// HPWH was heating last iteration and will continue to heat until the set point is reached
+			
 			HPPartLoadRatio = 1.0;
-
-			//         set up full air flow on DX coil inlet node
-			Node( DXCoilAirInletNode ).MassFlowRate = MdotAir;
-
-			//         set the condenser inlet node mass flow rate prior to calling the HPWH DX coil
-			Node( HPWaterInletNode ).MassFlowRate = MdotWater;
-			WaterThermalTank( WaterThermalTankNum ).SourceMassFlowRate = MdotWater;
```

Comment:
@rraustad The vast majority of this section of the diff was to clean up the code so we don't have duplicate blocks of code doing the same thing in two places. It's functionally equivalent to what was there before. I went to great lengths to verify that when I moved all this around it didn't change the calculations or results. So, yes 7028 does the exact same thing 6849 *and* 7002 did, but now in one line. The PLR of 1.0 is applied there if it started out in heating mode. Then it is corrected later using that false position solver and the residual calculations. 

I guess what I'm saying is that if the DXCoil isn't turning on when it should, it's not because of this change.